### PR TITLE
[LM-9367] DMFeed Sync Changes

### PR DESCRIPTION
**Pull request by arnavr15**
## Description

**Features**
- Added DMFeed Sync Changes

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [ ] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [ ] Label added
- [ ] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [ ] Description about the Feature/ Bugs added

### CODE REVIEW

File: ChatSX/screens/ChatRoom/index.tsx [hunk 1]
```diff
@@ -613,17 +651,43 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
...
+        ) {
+          const convToBeReplied = await myClient?.getConversation(
+            conversation.replyId,
+          );
+          if (convToBeReplied.length !== 0) {
+            DB_RESPONSE.conversationsData[i].replyConversationObject =
+              convToBeReplied[0];
+          }
+        }
+      }
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:682]: move this whole code to ROConvertor

File: ChatSX/screens/ChatRoom/index.tsx [hunk 2]
```diff
@@ -632,19 +696,60 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
...
+        if (keyA > keyB) return -1;
+        if (keyA < keyB) return 1;
+        return 0;
+      });
+      dispatch({
+        type: GET_CONVERSATIONS_SUCCESS,
+        body: {conversations: existingChatrooms},
+      });
+    }
+  };
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:731]: rename as per conversation

File: ChatSX/screens/ChatRoom/index.tsx [hunk 3]
```diff
@@ -1029,7 +1128,8 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
           navigation.goBack();
         } else {
           navigation.goBack();
-          await myClient?.deleteOneChatroom(chatroomID);
+          // Deleting the chatroom from realm in case of leaving the chatroom
+          await myClient?.deleteChatroom(chatroomID);
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:1132]: remove this query and add query to update follow status

File: ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx [hunk 4]
```diff
@@ -99,6 +106,157 @@ const DMFeed = ({navigation}: Props) => {
...
+  // Filtering DMs from GroupFeed Chatrooms
+  const filterDMs = (chatrooms: any) => {
+    let dmArr: any = [];
+    chatrooms.forEach((chatroom: any) => {
+      if (chatroom.type === 10) {
+        dmArr.push(chatroom);
+      }
+    });
+    return dmArr;
+  };
```
Comment (ishaan1607) [inline ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx:185]: move this for-loop to query

File: ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx [hunk 5]
```diff
@@ -99,6 +106,157 @@ const DMFeed = ({navigation}: Props) => {
...
+      const totalChatrooms = DB_RESPONSE?.chatroomsData;
+      for (let i = 0; i < totalChatrooms.length; i++) {
+        const chatroom = totalChatrooms[i];
+        const userData =
+          user?.id !== chatroom?.chatroomWithUserId
+            ? DB_RESPONSE?.userMeta[chatroom?.chatroomWithUserId]?.name
+            : DB_RESPONSE?.userMeta[chatroom?.userId]?.name;
+
+        DB_RESPONSE.chatroomsData[i].chatroomWithUserName = userData;
+      }
```
Comment (ishaan1607) [inline ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx:158]: move this part to ROConvertor

File: ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx [hunk 6]
```diff
@@ -99,6 +106,157 @@ const DMFeed = ({navigation}: Props) => {
...
+        ? 0
+        : parsedTimeStamp[0].maxTimeStamp;
+
+    const val = await syncChatroomAPI(page, minTimeStampNow, maxTimeStampNow);
+
+    const DB_RESPONSE = val?.data;
+
+    if (page === INITIAL_SYNC_PAGE && DB_RESPONSE?.chatroomsData.length !== 0) {
+      await myClient?.saveCommunity(DB_RESPONSE?.communityMeta[communityId]);
+    }
```
Comment (ishaan1607) [inline ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx:146]: move this code to separate file and use it here.

File: ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx [hunk 7]
```diff
@@ -99,6 +106,157 @@ const DMFeed = ({navigation}: Props) => {
...
+          if (keyA > keyB) return -1;
+          if (keyA < keyB) return 1;
+          return 0;
+        });
+        setRealmDMs(temp);
+      },
+      error: error => {
+        Alert.alert('Error observing chatrooms:', error);
+      },
+    });
```
Comment (ishaan1607) [inline ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx:240]: move this code to separate file

File: ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx [hunk 8]
```diff
@@ -99,6 +106,157 @@ const DMFeed = ({navigation}: Props) => {
...
+    return () => {
+      subscription.unsubscribe();
+    };
+  };
+
+  useEffect(() => {
```
Comment (ishaan1607) [inline ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx:246]: add comments

File: ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx [hunk 9]
```diff
@@ -99,6 +106,157 @@ const DMFeed = ({navigation}: Props) => {
...
+    listener();
+  }, [isFocused]);
+
+  useEffect(() => {
```
Comment (ishaan1607) [inline ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx:250]: add comments

File: ChatSX/screens/HomeFeed/Tabs/GroupFeed/index.tsx [hunk 10]
```diff
@@ -60,7 +60,18 @@ const GroupFeed = ({navigation}: Props) => {
...
+  // Filtering GroupFeed Chatrooms from DMs
+  const filterChatrooms = (chatrooms: any) => {
+    let chatroomArr: any = [];
+    chatrooms.forEach((chatroom: any) => {
+      if (chatroom.type !== 10) {
+        chatroomArr.push(chatroom);
+      }
+    });
+    return chatroomArr;
+  };
```
Comment (ishaan1607) [inline ChatSX/screens/HomeFeed/Tabs/GroupFeed/index.tsx:72]: move this to query

File: ChatSX/screens/HomeFeed/Tabs/GroupFeed/index.tsx [hunk 11]
```diff
@@ -162,14 +178,16 @@ const GroupFeed = ({navigation}: Props) => {
     });
     const subscription = chatroomObservable.subscribe({
       next: (updatedChatrooms: any) => {
-        updatedChatrooms.sort(function (a: any, b: any) {
+        const temp = filterChatrooms(updatedChatrooms);
+        // Sorting the updatedChatrooms based on updatedAt
+        temp.sort(function (a: any, b: any) {
```
Comment (ishaan1607) [inline ChatSX/screens/HomeFeed/Tabs/GroupFeed/index.tsx:183]: add sort in query

File: ChatSX/store/reducers/fileUploadReducer.ts [hunk 12]
```diff
@@ -34,11 +35,17 @@ export function fileUploadReducer(state = initialState, action: any) {
...
-        const res = await AsyncStorage.getItem('uploadingFilesMessages');
-        await AsyncStorage.setItem(
-          'uploadingFilesMessages',
-          JSON.stringify({...JSON.parse(res as any), ...dummyState}),
+        // const res = await AsyncStorage.getItem('uploadingFilesMessages');
+        // await AsyncStorage.setItem(
+        //   'uploadingFilesMessages',
+        //   JSON.stringify({...JSON.parse(res as any), ...dummyState}),
+        // );
+
```
Comment (ishaan1607) [inline ChatSX/store/reducers/fileUploadReducer.ts:43]: remove commented code

File: ChatSX/store/reducers/fileUploadReducer.ts [hunk 13]
```diff
@@ -58,12 +65,14 @@ export function fileUploadReducer(state = initialState, action: any) {
...
-        await AsyncStorage.setItem(
-          'uploadingFilesMessages',
-          JSON.stringify(obj),
-        );
+        // const res = await AsyncStorage.getItem('uploadingFilesMessages');
+        // let obj = removeKey(ID, JSON.parse(res as any));
+        // await AsyncStorage.setItem(
+        //   'uploadingFilesMessages',
+        //   JSON.stringify(obj),
+        // );
```
Comment (ishaan1607) [inline ChatSX/store/reducers/fileUploadReducer.ts:73]: remove commented code

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: ChatSX/screens/HomeFeed/Tabs/GroupFeed/index.tsx (hunk 11 above)
Comment (arnavr15) [reply ChatSX/screens/HomeFeed/Tabs/GroupFeed/index.tsx]: okay

File: ChatSX/screens/HomeFeed/Tabs/GroupFeed/index.tsx (hunk 10 above)
Comment (arnavr15) [reply ChatSX/screens/HomeFeed/Tabs/GroupFeed/index.tsx]: okay

File: ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx (hunk 5 above)
Comment (arnavr15) [reply ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx]: can't move this part to `ROConverter` as here I am using `user` which comes from Redux

File: ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx (hunk 4 above)
Comment (arnavr15) [reply ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx]: okay

**Pull request merged**